src/System.Net.Http/src/System/Net/Http/MessageProcessingHandler.cs

```diff
@@ -12,6 +13,22 @@ namespace System.Net.Http
 {
     public abstract class MessageProcessingHandler : DelegatingHandler
     {
+        // Private class used to capture the SendAsync state in
+        // a closure, while simultaneously avoiding a tuple allocation.
+        private class SendState : TaskCompletionSource<HttpResponseMessage>
+        {
+            public MessageProcessingHandler self;
+            public CancellationToken token;
```

Comment:
Whoops, I'd forgotten to add `readonly`. Good point.

Personally, I prefer to have public members inside private/internal classes because it's easier to refactor once you decide that class needs to be public, and it's also easier to distinguish the 'interface' from the implementation details.